src/main/docs/guide/httpServer/customArgumentBinding.adoc

```diff
@@ -0,0 +1,60 @@
+
+== Custom argument binding
+
+Micronaut uses an api:http.bind.binders.ArgumentBinderRegistry[ArgumentBinderRegistry] to look up api:core.bind.ArgumentBinder[ArgumentBinder]s
+capable of binding to the arguments in the controller methods. The default implementation will look for an annotation on
+the argument that is, itself, annotation with `@Bindable`. If one exists the argument binder registry will search for an
+argument binder that supports that annotation.
+
+If no fitting annotation is found Micronaut will try to find an argument binder that supports the argument's type.
+
+An api:core.bind.ArgumentBinder[ArgumentBinder] can be registered as a bean. In this case it is only present when a
+api:context.BeanContext[BeanContext] is initialized.
+
+Key interfaces are:
+
+=== AnnotatedRequestArgumentBinders
+* api:htp.bind.binders.AnnotatedRequestArgumentBinder[AnnotatedRequestArgumentBinder]
+
+- An interface for classes that bind an api:core.type.Argument[Argument] from an api:http.HttpRequest[HttpRequest] driven
+by an annotation.
+
+TIP: It is common to use api:io.micronaut.core.convert.ConversionService[ConversionService] to convert the data to the type of argument.
+
+=== TypedRequestArgumentBinders
+
+* api:http.bind.binders.TypedRequestArgumentBinder[TypedRequestArgumentBinder]
+
+- An interface for classes that bind an api:core.type.Argument[Argument] from an api:http.HttpRequest[HttpRequest] driven by
+the argument's type.
+
+Binders are called by the api:io.micronaut.http.annotation.Controller[Controller] once the request is received and the argument
```

Comment:
This paragraph is confusing to me. The filters aren't called by controllers. Both controllers and the binders are called by the server logic. I think the next paragraph is a great description, however I think an example should be given (the filters example). I would move the generic descriptions of argument binders above the specific information for typed and annotated otherwise it seems the generic information is specific to typed. The code samples need to be snippets and available in all languages and need to have tests to verify the code actually works.

Reply:
If I understood your recommendations correctly this new push should do what you suggested. Fun fact: I had never written Kotlin before this!